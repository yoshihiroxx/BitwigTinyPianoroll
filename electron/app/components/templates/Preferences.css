.container{
  background: #353535;
  color: #ffffff;
}
