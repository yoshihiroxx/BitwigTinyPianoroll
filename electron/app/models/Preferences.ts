import { Record } from 'immutable';
import KeyBinds from './KeyBinds';
import Theme from './Theme';
import GeneralPref from './GeneralPref';

// @todo write tests
export type PreferencesRecordType = {
  sequencer: {
    ppq: number;
  };
  general: GeneralPref;
  keyBinds: KeyBinds;
  theme: Theme;
  output: {
    defaultDirectory: string;
    midi: {
      formatType: number;
    };
  };
};

const PreferencesRecord = Record<PreferencesRecordType>({
  sequencer: {
    ppq: 960
  },
  general: new GeneralPref(),
  keyBinds: new KeyBinds(),
  theme: new Theme(),
  output: {
    defaultDirectory: './',
    midi: {
      formatType: 1
    }
  }
});

export default class Preferences extends PreferencesRecord {}
