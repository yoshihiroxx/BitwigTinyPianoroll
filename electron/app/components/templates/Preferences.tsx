import React from 'react';
import { Link } from 'react-router-dom';
import {
  Button,
  Tabs,
  Tab,
  TextField,
  Select,
  MenuItem,
  Toolbar,
  Typography,
  Paper,
  Grid
} from '@material-ui/core';
import {
  ThemeProvider,
  createMuiTheme,
  makeStyles,
  createStyles,
  styled
} from '@material-ui/core/styles';
import { display } from '@material-ui/system';
import PrefSelect from '../atoms/PrefSelect';
import PrefTabPanel from '../atoms/PrefTabPanel';
import PrefAppBar from '../atoms/PrefAppBar';

import styles from './Preferences.css';
import routes from '../../constants/routes.json';
import Preferences from '../../models/Preferences';
import PrefThemeTabPanel from '../molecules/PrefThemeTabPanel';
import Theme from '../../models/Theme';

type Props = {
  increment: () => void;
  incrementIfOdd: () => void;
  incrementAsync: () => void;
  decrement: () => void;
  counter: number;
  preferences: Preferences;
};

function a11yProps(index: any) {
  return {
    id: `simple-tab-${index}`,
    'aria-controls': `simple-tabpanel-${index}`
  };
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1
    },
    paper: {
      // padding: theme.spacing(2),
      textAlign: 'center',
      color: theme.palette.text.secondary
    },
    input: {
      margin: theme.spacing(2)
    }
  })
);

export default function PreferencesComponent(props: Props) {
  const { preferences } = props;

  const [value, setValue] = React.useState(0);

  const classes = useStyles(preferences.theme);

  const theme = createMuiTheme({
    palette: {
      type: 'dark',
      primary: {
        // light: will be calculated from palette.primary.main,
        main: '#F1831A'
        // dark: will be calculated from palette.primary.main,
        // contrastText: will be calculated to contrast with palette.primary.main
      },
      secondary: {
        light: '#0066ff',
        main: '#0044ff',
        // dark: will be calculated from palette.secondary.main,
        contrastText: '#ffcc00'
      }
      // Used by `getContrastText()` to maximize the contrast between
      // the background and the text.
      // contrastThreshold: 3,
      // Used by the functions below to shift a color's luminance by approximately
      // two indexes within its tonal palette.
      // E.g., shift from Red 500 to Red 300 or Red 700.
      // tonalOffset: 0.2
    },
    spacing: 8
  });

  const PrefButton = styled(Button)({
    background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
    border: 0,
    borderRadius: 3,
    boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
    color: 'white',
    height: 48,
    padding: '0 30px',
    margin: '23px'
  });

  const handleChange = (event: React.ChangeEvent<{}>, newValue: number) => {
    setValue(newValue);
  };

  return (
    <ThemeProvider theme={theme}>
      <PrefAppBar position="static">
        <Toolbar>
          <Typography variant="h6">Preferences</Typography>
        </Toolbar>
        <Tabs
          value={value}
          variant="fullWidth"
          onChange={handleChange}
          aria-label="simple tabs example"
          indicatorColor="primary"
          textColor="primary"
        >
          <Tab label="OSC" {...a11yProps(0)} />
          <Tab label="GENERAL" {...a11yProps(0)} />
          <Tab label="THEME" {...a11yProps(0)} />
        </Tabs>
      </PrefAppBar>
      <PrefThemeTabPanel
        currentIndex={value}
        index={2}
        state={preferences.theme}
      />
    </ThemeProvider>
  );
}
