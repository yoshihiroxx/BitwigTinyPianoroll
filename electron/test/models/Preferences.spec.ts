import Preferences from '../../app/models/Preferences';

describe('actions', () => {
  it('should increment should create increment action', () => {
    const pref = new Preferences();
    expect(pref).toMatchSnapshot();
  });
});
