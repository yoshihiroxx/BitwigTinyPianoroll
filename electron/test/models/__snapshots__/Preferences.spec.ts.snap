// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should increment should create increment action 1`] = `
Immutable.Record {
  "sequencer": Object {
    "ppq": 960,
  },
  "general": Immutable.Record {
    "clientMode": "bitwig",
    "isEnabled": true,
    "clientHost": "127.0.0.1",
    "clientPort": 3301,
    "serverHost": "127.0.0.1",
    "serverPort": 3330,
  },
  "keyBinds": Immutable.Record {
    "eraser": Object {
      "altKey": true,
      "codes": Array [
        18,
      ],
      "metaKey": false,
      "shiftKey": false,
    },
    "rect": Object {
      "altKey": false,
      "codes": Array [
        91,
        93,
      ],
      "metaKey": true,
      "shiftKey": false,
    },
    "copy": Object {
      "altKey": false,
      "codes": Array [
        67,
      ],
      "metaKey": true,
      "shiftKey": false,
    },
    "paste": Object {
      "altKey": false,
      "codes": Array [
        80,
      ],
      "metaKey": true,
      "shiftKey": false,
    },
    "pitchUp": Object {
      "altKey": false,
      "codes": Array [
        38,
      ],
      "metaKey": false,
      "shiftKey": false,
    },
    "pitchDown": Object {
      "altKey": false,
      "codes": Array [
        40,
      ],
      "metaKey": false,
      "shiftKey": false,
    },
    "increaseBeat": Object {
      "altKey": false,
      "codes": Array [
        39,
      ],
      "metaKey": false,
      "shiftKey": false,
    },
    "decreaseBeat": Object {
      "altKey": false,
      "codes": Array [
        37,
      ],
      "metaKey": false,
      "shiftKey": false,
    },
    "removeNotes": Object {
      "altKey": false,
      "codes": Array [
        8,
      ],
      "metaKey": false,
      "shiftKey": false,
    },
    "keepSelectionNotes": Object {
      "altKey": true,
      "codes": Array [
        18,
      ],
      "metaKey": false,
      "shiftKey": false,
    },
  },
  "theme": Immutable.Record {
    "pianoroll": Object {
      "background": Object {
        "alpha": 1,
        "value": 4473924,
      },
      "lines": Object {
        "horizontal": Object {
          "main": Object {
            "alpha": 0.7,
            "value": 16777215,
          },
          "sub": Object {
            "alpha": 0.4,
            "value": 16777215,
          },
        },
        "vertical": Object {
          "bar": Object {
            "alpha": 0.4,
            "value": 16777215,
          },
          "beat": Object {
            "alpha": 0.2,
            "value": 16777215,
          },
          "quarter": Object {
            "alpha": 0.2,
            "value": 16777215,
          },
        },
      },
      "notes": Object {
        "common": Object {
          "alpha": 1,
          "value": 3390361,
        },
        "drawing": Object {
          "alpha": 1,
          "value": 12320563,
        },
        "selection": Object {
          "alpha": 1,
          "value": 3399167,
        },
        "text": Object {
          "alpha": 0.4,
          "value": 0,
        },
      },
    },
  },
  "output": Object {
    "defaultDirectory": "./",
    "midi": Object {
      "formatType": 1,
    },
  },
}
`;
